app/src/services/cpk.ts

```diff
@@ -574,6 +662,79 @@ class CPKService {
       throw err
     }
   }
+
+  addGelatoSubmitTransaction = async (
+    collateralAmount: BigNumber,
+    priorCollateralAmount: BigNumber,
+    gelatoData: GelatoData,
+    gelato: GelatoService,
+    outcomeCount: number,
+    conditionalTokens: ConditionalTokenService,
+    conditionId: string,
+    collateralToken: Token,
+    marketMakerAddress: string,
+    account: string,
+  ) => {
+    const transactions = []
+
+    const { belowMinimum, minimum } = await this.isBelowGelatoMinimum(
+      collateralAmount,
+      collateralToken,
+      gelato,
+      priorCollateralAmount,
+    )
+
+    if (belowMinimum) {
+      logger.warn(`below gelato minimum ${minimum} ${collateralToken.symbol}, not using Gelato`)
+      return []
+    }
+
+    // Step 6: Enable Gelato Core as a module if not already done
+    const isGelatoWhitelistedModule = await gelato.isGelatoWhitelistedModule(this.cpk.address)
+    if (!isGelatoWhitelistedModule) {
+      const enableModuleData = await gelato.encodeWhitelistGelatoAsModule()
+      transactions.push({
+        to: this.cpk.address,
+        data: enableModuleData,
+      })
+    }
+
+    // Step 7: If automatic withdraw was selected, submit automatic Withdrawal Task to Gelato
+    const submitTaskData = await gelato.encodeSubmitTimeBasedWithdrawalTask({
+      gelatoData,
+      conditionalTokensAddress: conditionalTokens.address,
+      fpmmAddress: marketMakerAddress,
+      positionIds: await conditionalTokens.getPositionIds(outcomeCount, conditionId, collateralToken.address),
+      conditionId,
+      collateralTokenAddress: collateralToken.address,
+      receiver: account,
+    })
+
+    transactions.push({
+      to: gelato.addresses.gelatoCore,
+      data: submitTaskData,
+    })
+
+    return transactions
+  }
+
+  isBelowGelatoMinimum = async (
+    amount: BigNumber,
+    collateralToken: Token,
+    gelato: GelatoService,
+    priorAmount?: BigNumber,
+  ) => {
+    if (!priorAmount) {
+      priorAmount = Zero
+    }
+    const minDepositAmount = await gelato.minimumTokenAmount(collateralToken.address, collateralToken.decimals)
+    const depositAmount = Number(ethers.utils.formatUnits(amount, collateralToken.decimals))
+    const priorDepositAmount = Number(ethers.utils.formatUnits(priorAmount, collateralToken.decimals))
+    return {
+      belowMinimum: minDepositAmount > depositAmount + priorDepositAmount,
```

Comment:
I get same issue but it's not actually from this line (here `belowMinimum` is true if `minDepositAmount` is strictly greater than total deposited i.e. we are NOT below the minimum if we deposit greater or equal to the `minDepositAmount` so its correct). The issue is coming from rounding issue (`minDepositAmount` is truncated on the UI but if I print all the decimals it says 'at least 100.00000001 DAI')

Reply:
^Sorry, above comment a bit confusing. What I meant was: when I tested putting exactly 100 DAI in the deposit field, it did not enable gelato as I would expect since the wording is "at least." But that wasn't because this inequality is backwards, it's because the minimum threshold being compared was not actually 100 DAI but in reality 100.00000001 DAI